File: src/main/java/javasst/ast/JavaSstNode.java
package javasst.ast;

import ast.Node;
import javasst.JavaSstType;

/**
 * A Java SST {@link Node}.
 */
public class JavaSstNode extends Node<JavaSstType, JavaSstType> {

    /**
     * Create a new {@link Node}.
     */
    public JavaSstNode() {
        this(null, null);
    }

    /**
     * Create a new {@link Node}.
     *
     * @param clazz The class.
     */
    public JavaSstNode(final JavaSstType clazz) {
        this(clazz, null);
    }

    /**
     * Create a new {@link Node}.
     *
     * @param clazz The class.
     * @param type  The type.
     */
    public JavaSstNode(final JavaSstType clazz, final JavaSstType type) {
        super();
        super.setClazz(clazz);
        super.setType(type);
    }
}

File: src/main/scala/javasst/ast/JavaSstNode.scala
package javasst.ast

import javasst.ast.JavaSstNodeType.JavaSstNodeType
import javasst.parser.JavaSstParserObject

import compiler.ast.Node

/**
  * A Java SST [[compiler.ast.Node]].
  *
  * @author Julian Schlichtholz
  * @version 1.0.0
  */
class JavaSstNode(clazz: JavaSstNodeType = null, typ: JavaSstNodeType = null)
  extends Node[JavaSstNodeType, JavaSstNodeType](clazz, typ) {
  override type Self = JavaSstNode
  override type Object = JavaSstParserObject
}
